docs: reduce abstraction and make it easier to read (#48377)
freeCodeCamp/freeCodeCamp: docs/how-to-make-contribution-to-codebase.md

> Follow these guidelines for contribute to the codebase. This is highly recommended if you want to contribute regularly.

Reviewer: ```suggestion
Follow these guidelines to contribute to the codebase. This is highly recommended if you want to contribute regularly.
```

> Ignoring these steps will soil your copy which makes the contributing, maintaining, and reviewing processes difficult.

Reviewer 2: ```suggestion
Ignoring these steps may soil your copy which makes the contributing, maintaining, and reviewing processes difficult.
```

> ## Making Contributing To Codebase

Reviewer 2: ```suggestion
## Contributing to the Codebase
```

> You can now make changes to files and commit your changes to your clone of your fork, which you can prepare by reading [how to setup freecodecamp](how-to-setup-freecodecamp-locally.md).

Reviewer: ```suggestion
You can now make changes to files and commit your changes to your fork, which you can prepare by reading [how to set up freecodecamp](how-to-setup-freecodecamp-locally.md).
```

Follow these steps:

1. Validate that you are on the `main` branch:

   ```console
   git status
   ```

   You should get an output like this:

   ```console
   On branch main
   Your branch is up-to-date with 'origin/main'.

   nothing to commit, working directory clean
   ```

   If you got different message, then you aren't on main or your working directory isn't clean, resolve any outstanding files/commits and checkout `main`:

   ```console
   git checkout main
   ```

2. Sync the latest changes from the freeCodeCamp upstream `main` branch to your `main` fork branch:

   > [!WARNING]
   > If you have any outstanding pull requests that you made from the `main` branch of your fork, you will lose them at the end of this step.
   >
   > You should ensure your pull request is merged by a moderator before performing this step. To avoid this scenario, you should **always** work on a branch other than the `main`.

>    This step **will sync the latest changes** from the main repository of freeCodeCamp. It is important that you rebase your branch on top of the latest `upstream/main` as often as possible to avoid conflicts later.

Reviewer 2: "It is important that you rebase your branch on top of the latest `upstream/main` as often as possible to avoid conflicts later."

I didn't make a suggestion for this one. It feels like this line should be further down somewhere, but I didn't have a good place to suggest to add it.

Author: with some editing, this place seems fitting 🤔

   ```console
   git status
   ```

   Output:

   ```console
   On branch feat/documentation
   Your branch is up to date with 'upstream/feat/documentation'.

   Changes to be committed:
   (use "git reset HEAD <file>..." to unstage)

       modified:   CONTRIBUTING.md
       modified:   docs/README.md
       modified:   docs/how-to-setup-freecodecamp-locally.md
       modified:   docs/how-to-work-on-guide-articles.md
   ```

   Now, you can commit your changes with a short message like so:

   ```console
   git commit -m "fix: my short commit message"
   ```

   Some examples:

   ```md
   fix: add test for JavaScript - for loop step
   feat: add link for article for alexa skills
   ```

   Make a conventional commit message. This is a good practice As a developer, and you will be following standard practices.

   Some examples of conventional commit messages are:

   ```md
   fix: improve HTML step
   fix: fix build scripts for Travis-CI
   feat: add link article JavaScript hoisting
   docs: update contributing guidelines
   ```

   Keep these short, not more than 50 characters. You can always add additional information in the description of the commit message.

>    This does not take any additional time than an unconventional message like 'update file' or 'add index.md'

Reviewer 2: ```suggestion
   This does not take any more time than an unconventional message like 'update file' or 'add index.md'
```
